src/emitters/emitter.sqns.ts

```diff
@@ -382,27 +405,69 @@ export class SqnsEmitter implements IEmitter {
     const queueUrl = this.getQueueUrl(this.getQueueName(topic));
     const result = await this.sqsProducer.sendBatch(
       queueUrl,
-      messages.map((message) => {
-        return {
-          data: [message.data],
-          deduplicationId: message.deduplicationId,
-          eventName: topic.name,
-          messageAttributes: message.MessageAttributes,
-          messageGroupId: message.partitionKey || topic.name,
-          delay: message.delay || DEFAULT_MESSAGE_DELAY,
-          id: message.id,
-        };
-      })
+      this.getTransformedMessagesForQueue(topic.name, messages),
     );
-    return (result.Failed?.map((failed) => ({
-      id: failed.Id,
-      code: failed.Code,
-      message: failed.Message,
-      wasSenderFault: failed.SenderFault,
-    })) || []
+    return (
+      result.Failed?.map((failed) => ({
+        id: failed.Id,
+        code: failed.Code,
+        message: failed.Message,
+        wasSenderFault: failed.SenderFault,
+      })) || []
     );
   }
 
+  getBatchEmitPayload(
+    eventName: string,
+    messages: IBatchMessage[],
+    options?: IBatchEmitOptions
+  ): EmitBatchPayload {
+    const topic: Topic = {
+      name: eventName,
+      isFifo: !!options?.isFifo,
+      exchangeType: options?.exchangeType || ExchangeType.Fanout,
+      separateConsumerGroup: options?.consumerGroup,
+    };
+    if (topic.exchangeType === ExchangeType.Queue) {
+      const queueUrl = this.getQueueUrl(this.getQueueName(topic));
+      return this.sqsProducer.getBatchMessageRequests(
+        queueUrl,
+        this.getTransformedMessagesForQueue(topic.name, messages),
+      );
+    } else {
+      const topicArn = this.getTopicArn(this.getTopicName(topic));
+      return this.snsProducer.getBatchPublishInputs(
```

Comment:
```suggestion
      return this.snsProducer.getBatchPublishInput(
```

Reply:
Plural is more suitable here.

Reply:
I don't think so because it returns the interface BatchPublishInput which in itself contains the multiple messages